internal/web/context.go

```diff
@@ -0,0 +1,22 @@
+package web
+
+import (
+	"github.com/labstack/echo/v4"
+	"go.uber.org/zap"
+)
+
+type AppContext struct {
+	echo.Context
+	AppLogger *zap.Logger
+}
```

Comment:
The function CreateAppContext takes a logger but does not handle any potential nil values for the logger. You should add a check to ensure that the logger is non-nil before using it, or document that a nil logger might cause a panic later in the code.